packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -545,9 +545,13 @@ export class PipelineEditor extends React.Component<
     }
 
     app_data.runtime_image = propertySet.runtime_image;
-    app_data.outputs = propertySet.outputs;
-    app_data.env_vars = propertySet.env_vars;
-    app_data.dependencies = propertySet.dependencies;
+    app_data.outputs = propertySet.outputs.filter((x: any) => x !== undefined);
+    app_data.env_vars = propertySet.env_vars.filter(
+      (x: any) => x !== undefined
+    );
+    app_data.dependencies = propertySet.dependencies.filter(
+      (x: any) => x !== undefined
+    );
     app_data.include_subdirectories = propertySet.include_subdirectories;
     app_data.cpu = propertySet.cpu;
```

Comment:
Would the same apply here? We are seeing null values being saved when you initially set these to a value, and then try to delete the value from the node property (cpu, gpu, mem)

Reply:
I'm unsure, in the case of string array we wrote our own component to handle them (`StringArrayInput`) and that component caused this bug. Whereas iiuc we leverage a canvas component for the dropdown so I would argue that that is a separate issue than this.

Reply:
Yea, I would say that is a separate issue. I would consider storing `dependencies: ["one", null, null, "two", null, "three"]` a bug, but storing `cpu: null` more of an opinion. I think the same discussion could be had for when the array is empty should we store `dependencies: []` or just delete dependencies